### DISCUSSION TITLE
[Feature idea] File Preview Viewer in PCManFM

### DISCUSSION THREAD

**Discussion by mr360** [lxqt/lxqt#2548] [Ideas]
Hi all,

**Preface:** This is an idea that I am interested in working on, but before I do so I'd like to get some feedback on whether this is something the community also wants or will be open to. 

I'm thinking of adding functionality which would allow PCManFM to do the following:
- View PDF documents  (.pdf)
- View OpenDoc documents (.odt)
- View Images (jpeg, png, gif, webm)
- View videos (mp4)

![image](https://github.com/lxqt/lxqt/assets/6275600/9a9adc0f-a77b-4e27-8add-1930b7175b51)

**Comment by stefonarch** [comment]
I'm not a big friend of apps that do unrelated things, it remembers me of konqueror too much.
Filemanager are not for viewing documents or files IMHO.

But as a preview function it could be a nice feature indeed.

**Comment by tsujan** [comment]
In a nutshell, this should not be done.

Unlike KDE, LXQt doesn't clutter apps by making them do jobs that aren't related to them, not even as options. PCManFM-Qt should remain lightweight, and its code should remain manageable.

**Comment by mr360** [comment]
Thanks @stefonarch @tsujan for your input, 

I consider file previewing an integral aspect of file exploration. Instantly, I expect to glimpse the image or file contents, reserving further editing or processing for when I open it in a dedicated application. 

Nonetheless, I appreciate the rationale for maintaining lightweight and manageable code. If this feature could be implemented to meet coding requirements—remaining both lightweight and manageable—would it be likely to be integrated?

**Reply by tsujan** [reply]
> I consider file previewing an integral aspect of file exploration.

While respecting your opinion, I disagree. Another user might say the same thing about audio preview, video preview, etc. Of course, all of them are possible, but that's against LXQt's philosophy, which is based on Unix philosophy: "Write programs that do one thing and do it well."

> If this feature could be implemented to meet coding requirements—remaining both lightweight and manageable—would it be likely to be integrated?

Yes, but that's simply impossible ;)

**Comment by mr360** [comment]
Fair enough :-) Thanks for your input guys. I'll probably implement it for myself then -- because it's something I need and thus want to do.

**Reply by marcusbritanicus** [reply]
@mr360 I had implemented this in [NewBreeze](https://gitlab.com/marcusbritanicus/NewBreeze) long ago. Not quite the way you want, but you can use that as a start. 

And I totally agree with @tsujan. You can't make it lightweight and manageable.
